### PR TITLE
add .clang-format

### PR DESCRIPTION
**Pull request by joshinils**

### CODE REVIEW

**Comment by joshinils** [comment]
@philsupertramp i purposefully did not format any files as not to create potential merge-conflicts with the other open pull-requests.

this is the time to decide wether we keep the indent size of 2 or change it, i will probably format all file i touch going forward with the given clang-format file

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+UseTab: Never
+
+# jnl this may be shortened, but this will effect a lot of the formatting
+ColumnLimit: 200
+
+#jnl maybe even 0
+SpacesBeforeTrailingComments: 1
+
+# jnl again, not sure what this does
+ContinuationIndentWidth: 0
```
Comment (philsupertramp) [inline .clang-format:20]: i.e.
```
void foo(int a, 
         int b,
 here -> int c)
...
```

**Review by philsupertramp — APPROVED** [review]

File: .clang-format (hunk 1 above)
Comment (joshinils) [reply .clang-format:20]: no, from https://clang.llvm.org/docs/ClangFormatStyleOptions.html :
```cpp
int i =         //  VeryVeryVeryVeryVeryLongComment
  longFunction( // Again a long comment
    arg);
```

Comment (philsupertramp) [reply .clang-format:20]: yes, you are right, I used the value provided in the config file.
using
```
ContinuationIndentWidth: 2
```
will add two spaces in front of `int c`
```
void foo(int a, 
         int b,
 here -> ++int c)
...

Comment (philsupertramp) [reply .clang-format:20]: this rule is better described in a python linter [here](https://www.flake8rules.com/rules/E125.html)

**Pull request merged**